File: openmp_hello_world.cpp
#include <omp.h>
#include <cstdio>

int main(){
    
    int threads =-1;
    int my_thread=-1;
    
    #pragma omp parallel private(my_thread)
    {

        my_thread=omp_get_thread_num();
        threads=omp_get_num_threads();
        printf("Hello world from thread number %i of %i \n",my_thread,threads);
        
    }
    
    
    
}

File: openmp_hello_world.f90
program openmp_hello_world
use omp_lib
use omp_lib_kinds
implicit none

integer     ::threads,my_thread=-1
threads = omp_get_num_threads()

!$omp parallel private (my_thread)
my_thread = omp_get_thread_num()
threads = omp_get_num_threads()

print*, "Hello world from thread",my_thread,"of",omp_get_num_threads()

!$omp end parallel

endprogram
